src/installer/__main__.py

```diff
@@ -0,0 +1,67 @@
+"""CLI - run as python -m installer."""
+import argparse
+
+from . import install
+from .destinations import SchemeDictionaryDestination
+from .sources import WheelFile
+
+
+def main():
+    """Entry point for CLI."""
+    ap = argparse.ArgumentParser("python -m installer")
+    ap.add_argument("wheel_file", help="Path to a .whl file to install")
+
+    ap.add_argument(
+        "--interpreter", required=True, help="Interpreter path to be used in scripts"
+    )
+    ap.add_argument(
+        "--script-kind",
+        required=True,
+        choices=["posix", "win-ia32", "win-amd64", "win-arm", "win-arm64"],
+        help="Kind of launcher to create for each script",
+    )
+
+    dest_args = ap.add_argument_group("Destination directories")
+    dest_args.add_argument(
+        "--purelib",
+        required=True,
+        help="Directory for platform-independent Python modules",
+    )
+    dest_args.add_argument(
+        "--platlib",
+        required=True,
+        help="Directory for platform-dependent Python modules",
+    )
+    dest_args.add_argument(
+        "--headers", required=True, help="Directory for C header files"
+    )
+    dest_args.add_argument(
+        "--scripts", required=True, help="Directory for executable scripts"
+    )
+    dest_args.add_argument(
+        "--data", required=True, help="Directory for external data files"
+    )
+    args = ap.parse_args()
+
+    destination = SchemeDictionaryDestination(
+        {
+            "purelib": args.purelib,
+            "platlib": args.platlib,
```

Comment:
Don't these often go in the same place? Maybe we should have an argument to set both at the same time? I mean even when cross compiling we put target libraries all in the target site-packages. The host toolchain installation is done in a separate stage.

Reply:
If they do, then the caller can specify the same path.

As it stands, the (de facto) specifications/implementations allow specifying different paths for these, so this should also allow for them, to not make it impossible to do what the specification permits.

Reply:
Yeah, I was just suggesting an alternative convenience argument to set both to simplify the invocation a bit, not to remove the ability to set them separately.

Reply:
One of my suggestions above was to make one of these two optional and use the same location if it's not specified. That's a slightly simpler variant of having a third option to set both together.

Reply:
Let's make platlib optional, defaulting to the same value as purelib. :)